source/core/WorldGrid.hpp

```diff
@@ -193,6 +193,13 @@ namespace cse491 {
       return true;
     }
 
+    /// @brief Allows the system to replace any cell with a new cell type
+    /// @param grid 
+    /// @param newType 
+    void SetCell (GridPosition grid, int newType) {
```

Comment:
I don't think we need `SetCell` as a separate function.  The brackets operator (or the `At()` function) will take care of this.

Instead of (for example): `grid.SetCell(pos, in_type);`
You can already write: `grid[pos] = in_type;`

I think the current method is clear and concise, so I'd hesitate to add this new member function.